src/UI/Buyer/src/app/layout/header/header.component.ts

```diff
@@ -89,12 +92,21 @@ export class HeaderComponent implements OnInit, OnDestroy {
   }
 
   logout() {
-    this.ocTokenService.RemoveAccess();
-    if (this.appStateService.isAnonSubject.value) {
-      this.baseResolveService.resetUser();
-    } else {
-      this.router.navigate(['/login']);
-    }
+    this.appAuthService.logout();
+  }
+
+  clearSearchOnNavigate() {
+    this.activatedRoute.queryParams
+      .pipe(
+        filter((queryParams) => {
+          return typeof queryParams.search === 'undefined';
+        }),
+        takeWhile(() => this.alive)
+      )
+      .subscribe(() => {
+        const search = this.isMobile() ? this.mobileSearch : this.desktopSearch;
+        if (search) search.clearWithoutEmit();
```

Comment:
Can you wrap `search.clearWithoutEmit()` in brackets

Reply:
will do.